enigma-swing/src/main/java/cuchaz/enigma/gui/Gui.java

```diff
@@ -344,31 +394,44 @@ public void startRename(EditorPanel editor) {
 	}
 
 	public void showStructure(EditorPanel editor) {
-		this.structurePanel.showStructure(editor);
+		this.setRightPanel(RightPanel.Type.STRUCTURE, false);
+		((StructurePanel) this.getRightPanel()).showStructure(editor);
 	}
 
+	/**
+	 * Opens the Inheritance right panel and displays information for the provided editor's cursor reference.
+	 * @param editor the editor to extract the reference from
+	 */
 	public void showInheritance(EditorPanel editor) {
 		EntryReference<Entry<?>, Entry<?>> cursorReference = editor.getCursorReference();
 		if (cursorReference == null) return;
 
-		this.inheritanceTree.display(cursorReference.entry);
-		tabs.setSelectedIndex(1);
+		this.setRightPanel(RightPanel.Type.INHERITANCE, false);
+		((InheritanceTree) this.getRightPanel()).display(cursorReference.entry);
```

Comment:
`cursorReference.entry` is Nullable iirc

Reply:
I don't think so
this is the same way the previous code did it, and it never seemed to cause any issues

Reply:
Oh, I didn't see it. It's alright then